### PR TITLE
feat: make rfqm worker aware of its respective private key

### PR DESCRIPTION
**Pull request by alexkroeger**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

**Review by phil-ociraptor — APPROVED** [review]
lgtm!

File: src/runners/http_rfqm_service_runner.ts [hunk 1]
```diff
@@ -58,18 +62,35 @@ if (require.main === module) {
...
+        }
+        const workerPrivateKey = RfqBlockchainUtils.getPrivateKeyFromIndexAndPhrase(
+            META_TX_WORKER_MNEMONIC,
+            META_TX_WORKER_INDEX,
+        );
+        const privateWalletSubprovider = new PrivateKeyWalletSubprovider(workerPrivateKey);
+        const rpcProvider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
+        provider = RfqBlockchainUtils.createPrivateKeyProvider(rpcProvider, privateWalletSubprovider);
+    } else {
+        provider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
```
Comment (PirosB3) [inline src/runners/http_rfqm_service_runner.ts:92]: Small suggestion: Why not factor this out of the if/else loop? it looks like it's duplicated

File: src/utils/rfq_blockchain_utils.ts [hunk 2]
```diff
@@ -21,7 +24,8 @@ export class RfqBlockchainUtils {
     public static getPrivateKeyFromIndexAndPhrase(mnemonic: string, index: number): string {
         const hdNode = HDNode.fromMnemonic(mnemonic).derivePath(this._getPathByIndex(index));
 
-        return hdNode.privateKey;
+        // take '0x' off
+        return hdNode.privateKey.substring(2);
```
Comment (PirosB3) [inline src/utils/rfq_blockchain_utils.ts:28]: why are we stripping `0x` off

**Review by PirosB3 — APPROVED** [review]

**Comment by PirosB3** [comment]
Small question, if there is an exception anywhere in the code, is there a risk that the private keys may be exposed in the stack trace?

File: src/utils/rfq_blockchain_utils.ts (hunk 2 above)
Comment (alexkroeger) [reply src/utils/rfq_blockchain_utils.ts:28]: I believe the privateKeyProvider just expects the hex characters

**Pull request merged**